zipline/data/data_portal.py

```diff
@@ -0,0 +1,475 @@
+            'open': {},
+            'high': {},
+            'low': {},
+            'close': {},
+            'volume': {},
+            'sid': {},
+            'dt': {},
+        }
+
+        self._adjustment_reader = adjustment_reader
+
+        # caches of sid -> adjustment list
+        self._splits_dict = {}
+        self._mergers_dict = {}
+        self._dividends_dict = {}
+
+        # Cache of sid -> the first trading day of an asset, even if that day
+        # is before 1/2/2002.
+        self._asset_start_dates = {}
+        self._asset_end_dates = {}
+
+        # Handle extra sources, like Fetcher.
+        self._augmented_sources_map = {}
+        self._extra_source_df = None
+
+        self._sim_params = sim_params
+        if self._sim_params is not None:
+            self._data_frequency = self._sim_params.data_frequency
+        else:
+            self._data_frequency = "minute"
+
+        self.MINUTE_PRICE_ADJUSTMENT_FACTOR = 0.001
+
+        self._equity_daily_reader = equity_daily_reader
+        self._equity_minute_reader = equity_minute_reader
+        self._future_daily_reader = future_daily_reader
+        self._future_minute_reader = future_minute_reader
+
+    def _open_minute_file(self, field, asset):
+        sid_str = str(int(asset))
+
+        try:
+            carray = self._carrays[field][sid_str]
+        except KeyError:
+            carray = self._carrays[field][sid_str] = \
+                self._get_ctable(asset)[field]
+
+        return carray
+
+    def _get_ctable(self, asset):
+        sid = int(asset)
+
+        if isinstance(asset, Future):
+            if self._future_minute_reader.sid_path_func is not None:
+                path = self._future_minute_reader.sid_path_func(
+                    self._future_minute_reader.rootdir, sid
+                )
+            else:
+                path = "{0}/{1}.bcolz".format(
+                    self._future_minute_reader.rootdir, sid)
+        elif isinstance(asset, Equity):
+            if self._equity_minute_reader.sid_path_func is not None:
+                path = self._equity_minute_reader.sid_path_func(
+                    self._equity_minute_reader.rootdir, sid
+                )
+            else:
+                path = "{0}/{1}.bcolz".format(
+                    self._equity_minute_reader.rootdir, sid)
+
+        return bcolz.open(path, mode='r')
+
+    def get_spot_value(self, asset, field, dt=None):
+        """
+        Public API method that returns a scalar value representing the value
+        of the desired asset's field at either the given dt, or this data
+        portal's current_dt.
+
+        Parameters
+        ---------
+        asset : Asset
+            The asset whose data is desired.gith
+
+        field: string
+            The desired field of the asset.  Valid values are "open",
+            "open_price", "high", "low", "close", "close_price", "volume", and
+            "price".
+
+        dt: pd.Timestamp
```

Comment:
> An alternative may be updating the current time in the sid data view/proxy.

That seems like it'd be cleaner.  IMO the DataPortal shouldn't know anything about the context of the running simulation.